tests/src/python/test_qgsgeometry.py

```diff
@@ -2334,6 +2334,18 @@ def testReshape(self):
         wkt = g.asWkt()
         assert compareWkt(expWkt, wkt), "testReshape failed: mismatch Expected:\n%s\nGot:\n%s\n" % (expWkt, wkt)
 
+        # Test reshape a polygon with a line starting or ending at the polygon's first vertex, no change expexted
+        g = QgsGeometry.fromWkt('Polygon ((0 0, 1 0, 1 1, 0 1, 0 0))')
+        expWkt = g.asWkt()
+        g.reshapeGeometry(QgsLineString([QgsPoint(0, 0), QgsPoint(-1, -1)]))
+        assert compareWkt(g.asWkt(), expWkt), "testReshape failed: mismatch Expected:\n%s\nGot:\n%s\n" % (expWkt, wkt)
```

Comment:
Better use `self.assertTrue`, I know there's a lot missing in this file, but it's good practice to use the nose2 methods.

Reply:
There are 551 other occurrences of `assert` instead of `self.assertTrue` in this file. Should I change them too and issue another PR?

Reply:
Migrating them would be great, but many will better be switched to `self.assertEqual`, `self.assertNotNone` and other methods, so it's a bit more than a trivial search and replace.